File: works/2331.py
A, P = input().split()
A = int(A)
P = int(P)

def Cal(n):
    sum = 0
    while n > 0:
        sum += pow(n%10, P)
        n = n // 10

    return sum


cnt = 1
# dictionary
m = {}
m[A] = cnt
cnt += 1

while True:
    A = Cal(A)
    if A in m:
        print(m[A]-1)
        break
    m[A] = cnt
    cnt += 1

File: works/2331.kt
import java.util.*;

var A = 0
var P = 0

fun Cal(n: Int): Int {
    var sum = 0
    var _n = n
    while(_n > 0)
    {
        var res = Math.pow((_n%10).toDouble(), P.toDouble())
        sum += res.toInt()
        _n /= 10
    }
    return sum
}

fun main(args: Array<String>)
{
    val sc = Scanner(System.`in`)
    var tmp = sc.nextLine()
    var s_tmp = tmp.split(" ")
    A = s_tmp[0].toInt()
    P = s_tmp[1].toInt()

    var m = HashMap<Int, Int>()
    var cnt = 1
    m.put(A, cnt++)

    while(true)
    {
        A = Cal(A)
        //println(A)
        if(m.containsKey(A))
        {
            var a = m.get(A)?.minus(1)
            print(a)
            break
        }
        m.put(A, cnt++)
    }
}